### ISSUE TITLE
How index_select part tensor value in a particular dimension in Pytorch？

### ISSUE THREAD

**Issue by qlwang25** [pytorch/pytorch#9808]
## Issue description
The index_select function can not personalize for each sample.
for example:
```
a shape size (B, L, H)
B batch_size=2,  L time_step=3, H dim=4  
>>> a
tensor([[[0, 1, 2, 3],
         [4, 5, 6, 7],
         [8, 9, 10, 11]],

        [[12, 13, 14, 15],
         [16, 17, 18, 19],
         [20, 21, 22, 23]]])

if use index_select function to reverse time_step:
>>> indices = torch.arange(3-1, -1, -1)
>>> indices
tensor([2.,  1.,  0.])
>>> torch.index_select(a, 1, indices.long())
tensor([[[8, 9, 10, 11],
         [4, 5, 6, 7],
         [0, 1, 2, 3]],

        [[20, 21, 22, 23],
         [16, 17, 18, 19],
         [12, 13, 14, 15]]])

But if given the length tensor[2, 1],
How to  reverse time_step and personalize for each sample?
In other words, How to get the following results?
tensor([[[4, 5, 6, 7],
         [0, 1, 2, 3],
         [8, 9, 10, 11]],

        [[12, 13, 14, 15],
         [16, 17, 18, 19],
         [20, 21, 22, 23]]])

```
The first sample is only reversed to a length of 2;
The second sample is only reversed to a length of 1;

I want to know if pytorch has such a method.
or 
please tell me how to do it?

```
```

**Comment by soumith** [comment]
please use https://discuss.pytorch.org for questions. (thank you :) )
